### PR TITLE
[PVR] PVRManager: Factor out playback state functionality into its own class

### PR DESCRIPTION
**Pull request by ksooo**
Another PVR refactoring PR. Idea is decoupling and separation of concerns.

In the distant future PVR shall no more know anything about any application playback states. So, this PR is a step into this direction.

Runtime-tested on macOS and Android, latest Kodi master.

### CODE REVIEW

File: xbmc/pvr/PVRPlaybackState.h [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingChannel(const std::shared_ptr<CPVRChannel>& channel) const;
+
+  /*!
+   * @brief Check if the given recording is playing.
+   * @param recording The recording to check.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingRecording(const std::shared_ptr<CPVRRecording> &recording) const;
```
Comment (Rechi) [inline xbmc/pvr/PVRPlaybackState.h]: > Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types

File: xbmc/pvr/PVRPlaybackState.cpp [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+using namespace PVR;
+
+class CPVRPlaybackState::CLastWatchedUpdateTimer : public CTimer, private ITimerCallback
+{
+public:
+  explicit CLastWatchedUpdateTimer(CPVRPlaybackState& state,
+                                   const std::shared_ptr<CPVRChannel>& channel,
+                                   const CDateTime& time)
+  : CTimer(this), m_state(state), m_channel(channel), m_time(time) {}
```
Comment (Rechi) [inline xbmc/pvr/PVRPlaybackState.cpp]:
```suggestion
    : CTimer(this)
    , m_state(state)
    , m_channel(channel)
    , m_time(time)
  {
  }
```

**Review by Rechi — DISMISSED** [review]
If you prefer the style you used in this PR, crate a PR with documentation and clang-format configuration changes.

File: xbmc/pvr/PVRPlaybackState.h [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  /*!
+   * @brief Return the channel that is currently playing.
+   * @return The channel or NULL if none is playing.
+   */
+  std::shared_ptr<CPVRChannel> GetPlayingChannel() const;
+
+  /*!
+   * @brief Return the recording that is currently playing.
+   * @return The recording or NULL if none is playing.
```
Comment (phunkyfish) [inline xbmc/pvr/PVRPlaybackState.h]: nullptr?

**Review by phunkyfish — COMMENTED** [review]
Outside of the left align from Rechi it looks good. Not sure if my comments originate from NULL or nullptr so maybe the comment not so important. Up to you.

File: xbmc/pvr/PVRPlaybackState.h (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/PVRPlaybackState.h]: Oversight on my end. Will change that.

File: xbmc/pvr/PVRPlaybackState.h [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  /*!
+   * @brief Check if the given epg tag is playing.
+   * @param epgTag The tag to check.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingEpgTag(const std::shared_ptr<CPVREpgInfoTag>& epgTag) const;
+
+  /*!
+   * @brief Return the channel that is currently playing.
+   * @return The channel or NULL if none is playing.
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/pvr/PVRPlaybackState.h]: sure, thanks.

File: xbmc/pvr/PVRPlaybackState.h (hunk 3 above)
Comment (ksooo) [reply xbmc/pvr/PVRPlaybackState.h]: yep

File: xbmc/pvr/PVRPlaybackState.h [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+  /*!
+   * @brief Return the recording that is currently playing.
+   * @return The recording or NULL if none is playing.
+   */
+  std::shared_ptr<CPVRRecording> GetPlayingRecording() const;
+
+  /*!
+   * @brief Return the epg tag that is currently playing.
+   * @return The tag or NULL if none is playing.
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/pvr/PVRPlaybackState.h]: yo

**Comment by ksooo** [comment]
> If you prefer the style you used in this PR, crate a PR with documentation and clang-format configuration changes.

 For the records: I do not accept any clang-format stuff as a blocker for any PR. Team Kodi developers have no common agreement or at least consensus on the rule set you're trying to enforce here by blocking a PR.

I will incorporate all requested changes except the ctor initlist reformat stuff and merge as soon as Jenkins is happy.

File: xbmc/pvr/PVRPlaybackState.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/PVRPlaybackState.cpp]: no

**Comment by Rechi** [comment]
> Team Kodi developers have no common agreement or at least consensus on the rule set 

That's the reason I'm asking you to submit a PR with your suggestions. Otherwise everyone wants to keep his preferred code style.

**Comment by ksooo** [comment]
Sigh. Not worth the fight. I have better things to do. I will just obey the rules.

**Comment by ksooo** [comment]
Here we go. All (!) requested changes are in. Please approve or comment on other findings.

**Pull request merged**